core/src/test/scala/pureconfig/ConfigReaderFailureLocationSuite.scala

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package pureconfig
+
+import com.typesafe.config.ConfigFactory
+
+import org.scalatest._
+import pureconfig.error._
+
+/**
+ * Suite of tests related to the location of ConfigValues that raised failures.
+ */
+class ConfigReaderFailureLocationSuite extends FlatSpec with Matchers with EitherValues {
+  "Loading configuration from files" should "show proper error locations when loading a single file" in {
+    import pureconfig.syntax._
+    case class Conf(a: Int, b: String, c: Int)
+
+    val workingDir = getClass.getResource("/").getFile
+    val file = "conf/error1/a.conf"
+    val conf = ConfigFactory.load(file).root()
+
+    val failures1 = conf.get("conf").to[Conf].left.value.toList
+    failures1 should have size 2
+    failures1 should contain(KeyNotFound("a", Some(ConfigValueLocation(workingDir + file, 0))))
+    failures1 should contain(CannotConvert(
+      "hello",
+      "Int",
+      """java.lang.NumberFormatException: For input string: "hello"""",
+      Some(ConfigValueLocation(workingDir + file, 2))))
+
+    val failures2 = conf.get("other-conf").to[Conf].left.value.toList
+    failures2 should have size 3
+    failures2 should contain(KeyNotFound("a", Some(ConfigValueLocation(workingDir + file, 6))))
+    failures2 should contain(KeyNotFound("b", Some(ConfigValueLocation(workingDir + file, 6))))
+    failures2 should contain(CannotConvert(
+      "hello",
+      "Int",
+      """java.lang.NumberFormatException: For input string: "hello"""",
+      Some(ConfigValueLocation(workingDir + file, 8))))
+  }
+
+  it should "show proper error location when loading from multiple files" in {
+    import pureconfig.syntax._
+    case class Conf(a: Int, b: String, c: Int)
+
+    val workingDir = getClass.getResource("/").getFile
+    val file1 = "conf/error2/a.conf"
```

Comment:
Suggestion: Rename the `error2` and `error1` parts of the paths these test files are in to something like `errorLocationReporting` to try to tie them to their purpose.